public/utils/contextMenu/getActions.tsx

```diff
@@ -1,26 +1,31 @@
 import React from 'react';
 import { i18n } from '@osd/i18n';
-import { EuiIconType } from '@elastic/eui/src/components/icon/icon';
+import { EuiIconType } from '@elastic/eui';
 import { toMountPoint } from '../../../../../src/plugins/opensearch_dashboards_react/public';
 import { Action } from '../../../../../src/plugins/ui_actions/public';
 import { createADAction } from '../../action/ad_dashboard_action';
 import AnywhereParentFlyout from '../../components/FeatureAnywhereContextMenu/AnywhereParentFlyout';
 import { Provider } from 'react-redux';
 import { CoreServicesContext } from '../../../public/components/CoreServices/CoreServices';
 import configureStore from '../../redux/configureStore';
-import DocumentationTitle from '../../../public/components/FeatureAnywhereContextMenu/DocumentationTitle/containers/DocumentationTitle';
-import { AD_DOCS_LINK } from '../constants';
+import DocumentationTitle from '../../components/FeatureAnywhereContextMenu/DocumentationTitle/containers/DocumentationTitle';
+import { AD_DOCS_LINK, APM_TRACE } from '../constants';
+import { CoreSetup } from '../../../../../src/core/public';
 
 // This is used to create all actions in the same context menu
 const grouping: Action['grouping'] = [
   {
     id: 'ad-dashboard-context-menu',
     getDisplayName: () => 'Anomaly Detector',
-    getIconType: () => 'apmTrace',
+    getIconType: () => APM_TRACE,
   },
 ];
 
-export const getActions = ({ core, plugins }) => {
+interface GetActionsProps {
+  core: CoreSetup;
+}
```

Comment:
`core` should also be consumed as a service similar to `embeddable`, so it doesn't need to be passed as props. If it's some specific service within `core`, that is even better. (e.g., if only notifications service from core is needed, can do something like `setNotifications(core.notifications)`, and then access via `getNotifications()`)

Reply:
tried to do something like this `const startServices = createStartServicesGetter(core.getStartServices);` in the plugin.ts file to avoid passing core. However, I don't find a getter function for it - https://github.com/search?q=org%3Aopensearch-project%20createStartServicesSetter&type=code

Reply:
What service do you need from core? It should be exposed and you can just set that service of core.

Reply:
removed passing core as a prop, added Overlays setter and getter